### ISSUE TITLE
Provide a IHealthCheckPublisher  which writes result to log

### ISSUE THREAD

**Issue by thompson-tomo** [dotnet/aspnetcore#53670]
## Summary

I want to have a Health publisher which will write to the Standard log the results of the health check.

## Motivation and goals

Currently I develop a lot of on-premise software which doesn't benefit from centralised monitoring & logging. With this change I will be able to collect logs from site & see what the health has been for a period of time. A key thing would be to ensure appropriate log levels are used

## In scope

- Release of new nuget package Microsoft.Extensions.Diagnostics.HealthChecks.Publishers.Logger

## Out of scope

Supporting any logger other Microsoft.Extensions.Logger

## Examples

The idea is to make it as simple for developer to enable and as such the idea would be:
```
builder.Services.AddHealthPublisherLogger()
```
Obviously it should be possible to define the HealthCheckPublisherOptions just like if it was explicitly added.

**Comment by davidfowl** [comment]
Why does this need to be in the box? Why not write one for your on prem software?

**Comment by thompson-tomo** [comment]
Writing one is not the issue, I am more than happy to write it & then it submit it as a PR etc. I am more thinking for key Microsoft libraries ie logger, System.DiagnosticSource should the appropriate publisher be provided in a similar manner to configuration, logging or do we simply rely on third parties to manage it in it's entirety.

**Comment by thompson-tomo** [comment]
For instance with the System.Diagnostic publisher it would then facilitate aspire to visualise the result of the health checks.

**Comment by davidfowl** [comment]
If there were a set of concrete scenarios then we would add this. I'll leave this open but I don't see a reason to add this to the framework.

**Comment by thompson-tomo** [comment]
So the scenarios, I see would be Aspire & OTEL more broadly as they could utilise the metrics such a publisher would have access to & would be able to provide.

At the same time having a publisher writing to logs is useful not just for my case but also for general debugging/development.

**Comment by davidfowl** [comment]
Right, if that how aspire choses to implement health checks (which isn't the plan right now), that would be a +1, but we don't have a logging publisher as a scenario for health checks.

**Comment by thompson-tomo** [comment]
I have reworked the description to make it more descriptive and it now encompasses the 2 use cases with the scenarios for each mentioned. Obviously Aspire would fall under the DiagnosticSource option.
